### PR TITLE
[terra-list] Fixed Heading levels and keyboard navigation

### PR DESCRIPTION
**Pull request by saket2403**
### Summary

Changes

- Home Key focuses on first list item.
- End Key focuses on last list item.
- Section Headers to start at heading level 2

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-list/src/ListSection.jsx [hunk 1]
```diff
@@ -22,9 +22,9 @@ const propTypes = {
    */
   isCollapsible: PropTypes.bool,
   /**
-   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`.
+   * Optionally sets the heading level. One of `2`, `3`, `4`, `5`, `6`.
    */
-  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+  level: PropTypes.oneOf([2, 3, 4, 5, 6]),
```
Comment (sugan2416) [inline packages/terra-list/src/ListSection.jsx:27]: Update defaultProps.level to 2

Comment (saket2403) [reply packages/terra-list/src/ListSection.jsx:27]: Done [c8c28cc](https://github.com/cerner/terra-core/pull/3748/commits/c8c28cc0b0a77f76f4fb59b45af2eb150b139ac6)

**Review by sugan2416 — APPROVED** [review]

File: packages/terra-list/src/ListSection.jsx (hunk 1 above)
Comment (ShettyAkarsh) [inline packages/terra-list/src/ListSection.jsx:27]: is there a reason we are removing default header as one ?

Comment (saket2403) [reply packages/terra-list/src/ListSection.jsx]: As per UX recommendation level 1 should be removed. So to make it passive I have internally converted the h1 to h2 and reverted this change and added apt descriprtion

File: packages/terra-core-docs/src/terra-dev-site/test/list/SingleSelectList.test.jsx [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
   {
-    title: 'Item 3',
+    title: 'Clark Kent',
     key: 'unique-3',
   },
   {
-    title: 'Item 4',
-    key: 'unique-4',
+    title: 'Anne White',
+    key: 'unique-4  ',
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/test/list/SingleSelectList.test.jsx:23]: nit: do we need space here ??

File: packages/terra-list/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Changed
+  * Fixed heading level for list sections.
```
Comment (supreethmr) [inline packages/terra-list/CHANGELOG.md:6]:
```suggestion
  * Removed heading level `1` for list sections.
```

File: packages/terra-list/src/ListSection.jsx [hunk 4]
```diff
@@ -23,6 +23,9 @@ const propTypes = {
   isCollapsible: PropTypes.bool,
   /**
    * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`.
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue)
+   * Do Not use level as 1 as there should be only one H1 in a page.
+   * The option to add H1 will be removed in the next MVB
```
Comment (supreethmr) [inline packages/terra-list/src/ListSection.jsx:28]:
```suggestion
   *  Replace value `1` of `level` prop with other level values. Do Not use level as 1 as there should be only one H1 in a page.
   * Heading value `1` will be converted as heading level `2` until it is removed in the next MVB release.
```

**Review by supreethmr — APPROVED** [review]

**Pull request merged**